partner_communication_switzerland/models/contracts.py

```diff
@@ -266,8 +266,33 @@ def _send_reminders_for_birthday_in_1day_or_2months(self):
         tomorrow = today + relativedelta(days=1)
         sponsorships_with_birthday_tomorrow = \
             self._get_sponsorships_with_child_birthday_on(tomorrow)
+
+        sponsorships_to_send_reminder = []
```

Comment:
You are doing the opposite of what is asked! You are sending the reminder to all of those who already wrote or did a gift :(
So I propose to rename this variable to `sponsorships_to_avoid` and make it a recordset instead of a list, because the method `_send_birthday_reminders` asks for a recordset and not a list of ids.
```suggestion
        sponsorships_to_avoid = self.env['recurring.contract']
```

Reply:
..............I must have some problems with this task...